### PR TITLE
Optimize app Dockerfile with npm cache

### PR DESCRIPTION
**Pull request by megahirt**
## Description

This PR is intended to be merged after Billy's service-worker PR.  It probably won't pass tests until it is merged.

This PR optimizes the build time of `docker-compose build app` in the following ways:

1. It utilizes an npm cache image to reduce/eliminate having to download npm packages from the internet.  As we update our dependencies, this npm cache image will slowly provide less performance value.  For now, it provides about a 50% speed increase in `npm install` time.
2. It remove unnecessary COPY lines that aren't used by webpack
3. It targets the COPY src/Site/views to only copy sass files

## Performance Metrics on Chris's machine from Thailand
`docker-compose build app`

### Before This PR

package.json file change 2:47
sass file, html file, typescript file change 0:47
PHP file change 0:03
TWIG file change 0:47

### After This PR
package.json file change **1:53**
sass file, html file, typescript file change 0:47
PHP file change 0:03
TWIG file change **0:09**

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have reviewed the title/description of this PR which will be used as the squashed PR commit message
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works

TODO:
- publish the npm-cache image to sillsdev so it's not published under megahirt.  At this time I don't have access to push and I need help getting push access.

### CODE REVIEW

File: docker/app/Dockerfile
Comment (megahirt) [inline docker/app/Dockerfile:14]: Once I know how to publish to sillsdev, we can change this away from the megahirt repo

File: docker/app/Dockerfile
Comment (megahirt) [inline docker/app/Dockerfile:29]: Removing the appManifest and service-worker lines will only work after #1247 is merged

Comment (longrunningprocess) [reply docker/app/Dockerfile:29]: I merged it, you may rebase now

File: docker/app/Dockerfile
Comment (longrunningprocess) [reply docker/app/Dockerfile:14]: I think this is promising, I'd like to suggest this be done via a GHA triggered off any push to `develop` and producing a `npm-cache-develop` tag.

Additionally, I'd like to consider/discuss the implications this may have when producing the "production" version of this image.

**Review by longrunningprocess — CHANGES_REQUESTED** [review]
Every little bit helps, thanks 🚀 

I believe this needs a little more discussion, let me know if/when we can talk.

File: docker/app/Dockerfile
Comment (megahirt) [reply docker/app/Dockerfile:14]: I don't see any issues with using this npm-cache in a production build.  It's a cache that npm should be able to intelligently use at it's discretion, but should not affect the product output or final app image in any way (thanks to multi-stage builds!).  The cache folder (`~/.npm`) will be generated either way.  Saving it as a docker image instead of having to rebuild it on every run should save time in both prod and dev image builds.  That's my thinking anyway.

Comment (megahirt) [reply docker/app/Dockerfile:14]: I like the idea of of publishing a GHA triggered npm-cache-develop image after we have proved out the concept.

Comment (longrunningprocess) [reply docker/app/Dockerfile]: yep, I was not thinking about it properly, thanks for reminding me this is transient ad won't impact our app integrity across versions.

**Comment by longrunningprocess** [comment]
@megahirt I had a moment to add the GHA and seed Docker Hub.  If you're good with these commits and there's nothing else you'd like to do I'm good with the merge.  Please feel free to squash-merge even though we've both worked in this one.

Additionally, we may want to think about our `make clean` targets because we have nothing in our `Makefile` at this point that would clean this new image...I might suggest a line in the `powerwash` target that will explicitly remove this image.

**Review by longrunningprocess — APPROVED** [review]
once this goes into `develop` it would be nice to ensure the button shows up for run the workflow on demand.

I made one suggestion related to cleaning things up when a dev requests a `powerwash` but I'll leave it to you whether you want to add the extra line or not.

**Comment by longrunningprocess** [comment]
:-D, nm, I went ahead and added the clean up did after I thought about it

**Pull request merged**
